desktop/src/app/components/navbar/warnings/modals/fix-outliers-modal.component.ts

```diff
@@ -53,8 +56,40 @@ export class FixOutliersModalComponent {
 
         this.projectDocument = await this.datastore.get('project');
         this.valuelist = await this.getValuelist(this.document, this.field);
+        this.foundDocuments = await this.datastore.find({
+            constraints: { ['outlierValues:contain']: this.outlierValue }
+        }, { includeResourcesWithoutValidParent: true }).then(res => res.documents);
+        this.countAffected = "";
+        this.documentsToChange = [];
+    }
+
+    public updateSelectedValue(event: string) {
+
+        this.selectedValue = event;
+        this.replaceAll = false;
     }
 
+    public async prepareReplaceAll() {
+        this.replaceAll = !this.replaceAll;
+
+        this.documentsToChange = [];
+
+        for (let document of this.foundDocuments) {
+            const category: CategoryForm = this.projectConfiguration.getCategory(document.resource.category);
+
+            for (let fieldName of Object.keys(document.warnings.outliers.fields)) {
+                const field: Field = CategoryForm.getField(category, fieldName);
+                if (!this.hasOutlierValue(document, field)) continue;
+                const valuelist: Valuelist = await this.getValuelist(document, field);
+                if (valuelist && equal(valuelist, this.valuelist)) {
+                    this.replaceValue(document, document.resource, field);
```

Comment:
The actual value replacement should not happen in this function but in "replaceMultiple". Then "replaceAll" does not have to be reset in "updateSelectedValue".

Reply:
Okay, I think it did it properly, but I am not sure I made the right decision there. Would be glad if you could check this again after I pushed to this pr.